README.md

```diff
@@ -2,27 +2,93 @@
-  By default, only alerts with the severity level of `Error` or security severity level of `Critical` or `High` will cause a pull request check failure, and a check will still succeed with alerts of lower severities.
+```yml
+slither-args: --filter-paths "./lib|./test" --exclude naming-convention
+```
+
+This means Slither is not run on the `lib` or `test` folders, and the `naming-convention` check is disabled.
+This `slither-args` field is where you can change the Slither configuration for your project.
+
+For more information on configuration Slither, see [the documentation](https://github.com/crytic/slither/wiki/Usage). For more information on configuring the slither action, see the [slither-action](https://github.com/crytic/slither-action) repo.
+
+### GitHub Code Scanning
+
+As mentioned, the Slither CI step is integrated with GitHub's [code scanning](https://docs.github.com/en/code-security/code-scanning) feature.
+This means when your jobs execute, you'll see two related checks:
+
+1. `CI / slither-analyze`
+2. `Code scanning results / Slither`
+
+The first check is the actual Slither analysis.
+You'll notice in the [`ci.yml`](.github/workflows/ci.yml) file that this check has a configuration of `fail-on: none`.
+This means this step will _never_ fail CI, no matter how many findings there are or what their severity is.
+Instead, this check outputs the findings to a SARIF file[^sarif] to be used in the next check.
+
+The second check is the GitHub code scanning check.
+The `slither-analyze` job uploads the SARIF report to GitHub, which is then analyzed by GitHub's code scanning feature in this step.
+This is the check that will fail CI if there are Slither findings.
+
+By default when you create a repository, only alerts with the severity level of `Error` will cause a pull request check failure, and checks will succeed with alerts of lower severities.
+However, you can [configure](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#defining-the-severities-causing-pull-request-check-failure) which level of slither results cause PR check failures.
+
+It's recommended to conservatively set the failure level to `Any` to start, and to reduce the failure level if you are unable to sufficiently tune Slither or find it to be too noisy.
+
+Findings are shown directly on the PR, as well as in your repo's "Security" tab, under the "Code scanning" section.
+Alerts that are dismissed are remembered by GitHub, and will not be shown again on future PRs.
+
+Note that code scanning integration [only works](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/setting-up-code-scanning-for-a-repository) for public repos, or private repos with GitHub Enterprise Cloud and a license for GitHub Advanced Security.
```

Comment:
This is pretty pricey, right? Is there a recommended alternative configuration that still includes Slither output for a closed repo in some way w/o having this service level from GitHub? Could definitely imagine someone wanting to use the template for an under-development project that wasn't yet open sourced.

Reply:
Looks like Enterprise pricing is $252/user/year, but Advanced Security is an add on and I can't find how much it costs, even on the [billing help](https://docs.github.com/en/billing/managing-billing-for-github-advanced-security/about-billing-for-github-advanced-security) page.

IMO the workaround would be to remove the code scanning integration and use slither normally. It has a triage mode which behaves similarly, e.g. "mark this finding as ignored and we won't report it again". So I think you'd probably be able to use that and commit the triage results. Figured I'd figure that one out more when we have the need, but maybe worth mentioning the workaround?

Reply:
Updated in 09cf570